internal/datastore/txn.go

```diff
@@ -91,9 +92,9 @@ type BasicTxn struct {
 var _ Txn = (*BasicTxn)(nil)
 
 // newTxnFrom returns a new Txn from the rootstore.
-func NewTxnFrom(rootstore corekv.TxnStore, id uint64, readonly bool) *BasicTxn {
+func NewTxnFrom(rootstore corekv.TxnStore, id uint64, readonly bool, chunkSize immutable.Option[int]) *BasicTxn {
 	rootTxn := rootstore.NewTxn(readonly)
-	multistore := NewMultistore(rootTxn)
+	multistore := NewMultistore(rootTxn, chunkSize)
```

Comment:
nitpick: rename to `multiStore`
```suggestion
	multiStore := NewMultistore(rootTxn, chunkSize)
```

Reply:
None of the stores use a camel case format. This was changed a while ago and I don't think we should reintroduce this convention here.